src/Analyzers/Core/CodeFixes/AliasAmbiguousType/AbstractAliasAmbiguousTypeCodeFixProvider.cs

```diff
@@ -32,40 +36,88 @@ public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+                        return Task.FromResult(document.WithSyntaxRoot(newRoot));
+                    },
+                    title));
+            }
+
+            context.RegisterCodeFix(
+                CodeAction.Create(
+                    string.Format(CodeFixesResources.Alias_ambiguous_type_0, diagnosticNode.ToString()),
+                    actions.ToImmutable(),
+                    isInlinable: true),
+                context.Diagnostics.First());
+        }
+
+        private static IEnumerable<ITypeSymbol> Sort(IEnumerable<ITypeSymbol> types, bool sortSystemFirst)
+        {
+            var typeToNames = new Dictionary<ITypeSymbol, ImmutableArray<string>>();
 
-                var options = await document.GetAddImportPlacementOptionsAsync(addImportService, context.GetOptionsProvider(), cancellationToken).ConfigureAwait(false);
+            return types.OrderBy((t1, t2) =>
+            {
+                var t1Names = GetNames(t1);
+                var t2Names = GetNames(t2);
 
-                var codeActionsBuilder = ImmutableArray.CreateBuilder<CodeAction>(symbolInfo.CandidateSymbols.Length);
-                foreach (var symbol in symbolInfo.CandidateSymbols.Cast<ITypeSymbol>())
+                for (int i = 0, n = Math.Min(t1Names.Length, t2Names.Length); i < n; i++)
                 {
-                    var typeName = symbol.Name;
-                    var title = GetTextPreviewOfChange(typeName, symbol);
-
-                    codeActionsBuilder.Add(CodeAction.Create(
-                        title,
-                        cancellationToken =>
-                        {
-                            var aliasDirective = syntaxGenerator.AliasImportDeclaration(typeName, symbol);
-                            var newRoot = addImportService.AddImport(compilation, root, diagnosticNode, aliasDirective, syntaxGenerator, options, cancellationToken);
-                            return Task.FromResult(document.WithSyntaxRoot(newRoot));
-                        },
-                        title));
+                    var t1Name = t1Names[i];
+                    var t2Name = t2Names[i];
+
+                    var comparer = i == 0 && sortSystemFirst ? SortSystemFirstComparer.Instance : StringComparer.Ordinal;
+
+                    var diff = comparer.Compare(t1Name, t2Name);
+                    if (diff != 0)
+                        return diff;
                 }
 
-                var groupingTitle = string.Format(CodeFixesResources.Alias_ambiguous_type_0, diagnosticNode.ToString());
-                var groupingCodeAction = CodeAction.Create(groupingTitle, codeActionsBuilder.ToImmutable(), isInlinable: true);
-                context.RegisterCodeFix(groupingCodeAction, context.Diagnostics.First());
+                return t1Names.Length - t2Names.Length;
+            });
+
+            ImmutableArray<string> GetNames(ITypeSymbol symbol)
```

Comment:
Any reason to not use `typeSymbol.ToNameDisplayString()` and compare the entire string directly (minus the StartsWith(System.)?  Comparison code might be a bit simpler that way

Reply:
Personally, I find string stuff harder to reason about. My weak brain :-D